src/snowflake/cli/plugins/init/commands.py

```diff
@@ -0,0 +1,238 @@
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional
+
+import typer
+import yaml
+from click import ClickException
+from snowflake.cli.api.commands.flags import (
+    NoInteractiveOption,
+    VariablesOption,
+    parse_key_value_variables,
+)
+from snowflake.cli.api.commands.snow_typer import SnowTyperFactory
+from snowflake.cli.api.constants import DEFAULT_SIZE_LIMIT_MB
+from snowflake.cli.api.output.types import (
+    CommandResult,
+    MessageResult,
+)
+from snowflake.cli.api.project.schemas.template import Template, TemplateVariable
+from snowflake.cli.api.secure_path import SecurePath
+from snowflake.cli.api.utils.rendering import get_template_cli_jinja_env
+
+# simple Typer with defaults because it won't become a command group as it contains only one command
+app = SnowTyperFactory()
+
+DEFAULT_SOURCE = "https://github.com/snowflakedb/snowflake-cli-templates"
+
+PathArgument = typer.Argument(
+    ...,
+    help="Directory to be initialized with the project. This directory must not already exist",
+    show_default=False,
+)
+TemplateOption = typer.Option(
+    None,
+    "--template",
+    help="which template (subdirectory of --template-source) should be used. If not provided,"
+    " whole source will be used as the template.",
+    show_default=False,
+)
+SourceOption = typer.Option(
+    default=DEFAULT_SOURCE,
+    help=f"local path to template directory or URL to git repository with templates.",
+)
+
+TEMPLATE_METADATA_FILE_NAME = "template.yml"
+
+
+def _fetch_local_template(
+    template_source: SecurePath, path: Optional[str], destination: SecurePath
+) -> SecurePath:
+    """Copies local template to [dest] and returns path to the template root.
+    Ends with an error of the template does not exist."""
+
+    template_source.assert_exists()
+    template_origin = template_source
+    if path:
+        template_origin = template_source / path
+    if not template_origin.exists():
+        raise ClickException(
+            f"Template '{path}' cannot be found under {template_source}"
+        )
+
+    template_origin.copy(destination.path)
+    return destination / template_origin.name
+
+
+def _fetch_remote_template(
+    url: str, path: Optional[str], destination: SecurePath
+) -> SecurePath:
+    """Downloads remote repository template to [dest],
+    and returns path to the template root.
+    Ends with an error of the template does not exist."""
+    from git import rmtree as git_rmtree
+
+    # TODO: during nativeapp refactor get rid of this dependency
+    from snowflake.cli.plugins.nativeapp.utils import shallow_git_clone
+
+    shallow_git_clone(url, to_path=destination.path)
+    if path:
+        template_root = destination / path
+    else:
+        # remove .git directory not to copy it to the template
+        template_root = destination
+        git_rmtree((template_root / ".git").path)
```

Comment:
Can you explain this part?

Reply:
This supports the case when whole repository is a single template. (compatibility with native app features)

Reply:
Sounds good, please add this comment to the code for future 🙆‍♂️

Reply:
done!